### ISSUE TITLE
Element Filters and Drawing Underlay panels not visible 

### ISSUE THREAD

**Issue by manuvarkey** [IfcOpenShell/IfcOpenShell#5233]
`Element Filters` and `Drawing Underlay` panels under `Drawings and Documents` used to be shown in the stable versions of BlenderBIM. However in latest version, these are missing.
 
![Screenshot from 2024-08-27 17-26-27](https://github.com/user-attachments/assets/6cbda8ee-4bdc-48c7-872e-bce7b94aa543)

I could find these classes in https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.8.0/src/bonsai/bonsai/bim/module/drawing/ui.py which is supposed to show these panels.

BIM_PT_element_filters
BIM_PT_drawing_underlay

**Comment by manuvarkey** [comment]
The IFC file attached in https://github.com/IfcOpenShell/IfcOpenShell/issues/5232#issuecomment-2312449448 has several drawings with element exclude filters applied for drawings. Although now it is not possible to edit these filters, the generated drawing correctly accounts for the exclude filter set in the IFC file.

**Comment by Andrej730** [comment]
Closed with https://github.com/IfcOpenShell/IfcOpenShell/commit/8309b13db0d8e65dddd66fb10193babfce213f42

**Comment by manuvarkey** [comment]
@Andrej730 After the commit, following error is displayed when include or exclude filter is selected without camera selected.

```
Python: Traceback (most recent call last):
  File "/home/manu/.config/blender/4.2/extensions/.local/lib/python3.11/site-packages/bonsai/bim/module/search/operator.py", line 50, in execute
    filter_groups = tool.Search.get_filter_groups(self.module)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/manu/.config/blender/4.2/extensions/.local/lib/python3.11/site-packages/bonsai/tool/search.py", line 28, in get_filter_groups
    return bpy.context.active_object.data.BIMCameraProperties.exclude_filter_groups
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'data'
```

The following  code is still referring to active object for getting camera properties.

https://github.com/IfcOpenShell/IfcOpenShell/blob/002c50e29de569d2781fa4bd8dcd6ead104d5f25/src/bonsai/bonsai/tool/search.py#L25-L28

**Comment by manuvarkey** [comment]
Related to the above, trying to add a blank include or exclude query as follows leads to error.

![Screenshot from 2024-08-27 22-51-28](https://github.com/user-attachments/assets/13be027d-4df2-4b34-81a1-89a97bf73e45)


```
Python: Traceback (most recent call last):
  File "/home/manu/.config/blender/4.2/extensions/.local/lib/python3.11/site-packages/bonsai/bim/ifc.py", line 386, in execute_ifc_operator
    result = getattr(operator, "_execute")(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/manu/.config/blender/4.2/extensions/.local/lib/python3.11/site-packages/bonsai/bim/module/drawing/operator.py", line 2692, in _execute
    bpy.ops.bim.activate_drawing(drawing=element.id(), camera_view_point=False)
  File "/snap/blender/5234/4.2/scripts/modules/bpy/ops.py", line 109, in __call__
    ret = _op_call(self.idname_py(), kw)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: Error: Python: Traceback (most recent call last):
  File "/home/manu/.config/blender/4.2/extensions/.local/lib/python3.11/site-packages/bonsai/bim/module/drawing/operator.py", line 1548, in execute
    core.activate_drawing_view(tool.Ifc, tool.Blender, tool.Drawing, drawing=drawing)
  File "/home/manu/.config/blender/4.2/extensions/.local/lib/python3.11/site-packages/bonsai/core/drawing.py", line 492, in activate_drawing_view
    drawing_tool.activate_drawing(camera)
  File "/home/manu/.config/blender/4.2/extensions/.local/lib/python3.11/site-packages/bonsai/tool/drawing.py", line 1873, in activate_drawing
    current_representation = tool.Geometry.get_active_representation(obj)
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/manu/.config/blender/4.2/extensions/.local/lib/python3.11/site-packages/bonsai/tool/geometry.py", line 363, in get_active_representation
    if obj.data and hasattr(obj.data, "BIMMeshProperties") and obj.data.BIMMeshProperties.ifc_definition_id:
       ^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'data'
Location: /snap/blender/5234/4.2/scripts/modules/bpy/ops.py:109

```

**Comment by Andrej730** [comment]
@manuvarkey fixed in the latest build:

>After the commit, following error is displayed when include or exclude filter is selected without camera selected.

Fixed in 4e1a956

> Related to the above, trying to add a blank include or exclude query as follows leads to error on clicking Save Include Filter / Save Exclude Filter.

Fixed in 1aed2a6 (was a separate issue that occured before the latest changes).
